javascript/ql/lib/semmle/javascript/frameworks/Nest.qll

```diff
@@ -523,10 +523,10 @@ module NestJS {
     result = tuple.getAPropertyWrite("useClass").getRhs()
     or
     exists(DataFlow::FunctionNode f |
-      f = tuple.getAPropertyWrite("useFactory").getRhs().getALocalSource() and
+      f = tuple.getAPropertyWrite("useFactory").getRhs().getAFunctionValue() and
```

Comment:
This change makes the predicate mutually recursive with call graph construction, which might cause some performance issues. I've started some performance evaluations to help determine how expensive it is.